PAPER.md

```diff
@@ -1,72 +1,84 @@
+In Presigning (Fig. 8), Round 2, step 2 `r_{i,j}` and `\hat{r}_{i,j}` should be drawn from `\mathbb{Z}_{N_i}`, and not `N_j`, judging by how they are used later.
 
+Fig. 7 (KeyRefresh): there is a variable `B` that is not introduced anywhere, but used in hashes - must be forgotten from the previous revision.
 
-# Protocol
+Fig. 7 (KeyRefresh): `srid` is not introduced anywhere - probably should be `rid`.
 
-Q: `\ell`, `ell_prime`, `\eps`, and corresponding intervals (\mathcal{I} and \mathcal{J}, see e.g. Fig. 7) - what are they equal to? It is mentioned that they will be "determined by the analysis", but I could not find the values specified. I found one mention of the range parameters (and the security parameter `m`): the caption of Table 2 in Appendix D. It gives `m = 80`, `\ell = \kappa`, `\ell^\prime = 5 \kappa`, `\eps = 2 \kappa`, where `\kappa` is the bit length of the curve order, so 256 for secp256k1. Also Fig. 28 in Section C.5 fixes the size of the RSA modulus as `log2(N) = 4 \ell + 2 \eps` (which gives 2048 bit). I'm currently using these as production parameters, but it seems weird that they're hidden in such obscure places.
+Fig. 7 (KeyRefresh), Output, 1. (c): should be `\hat{\psi}_{j,k}`, not `\psi_{j,k}`. `\psi_{j,k}` are `П^{fac}`, not `П^{sch}`.
 
-A (NM): Those are the recommended values.
+Fig. 8 (Presigning): the order of public parameters for `П^{elog}` is different from the one in Fig. 23. For example, in Round 2, 2a), the order is `\Gamma_i, g, B_{i,1}, B_{i,2}, Y_i`, but it should be `B_{i,1}, B_{i,2}, Y_i, \Gamma_i, g`.
 
----
+Fig. 9, Output potentially needs `D_{k,j}` for all `k, j` (`k != j`) to calculate `D_j`. But, with the messages described in Fig. 8, a party `i` would only have `D_{j,i}`, `j!=i` (the ones it created in Round 2), and `D_{i,j}`, `j!=i` (the ones it received from other nodes). How are we supposed to obtain the rest? One option is to echo-broadcast (echo, so that they could be used to generate a verifiable evidence, too) all `D_{j,i}` and `F_{j,i}` in Round 2 instead of sending each `D_{j,i}` and `F_{j,i}` to the corresponding node `j`. Same goes for `\hat{D}` and `\hat{F}`.
 
-Q: In Fig. 5 the initial commitment in the Schnorr proof (`A_i`) is sent twice (in R2 and then R3, as a part of `psi_i`), and then the equality of those two values is checked in Output, right before checking the proof itself. But the commitment value is used right after that in the `vrfy` step, so if it doesn't match the rest of the proof, the verification will fail. So is there any security reason for sending the commitment twice? (I understand that it does constitute a minor performance optimization)
+The above item leads to another problem. If those values are indeed echo-broadcasted, what malicious actions of one node can lead to passing the `П^{dec}` check on other nodes, but failing one of the `П^{aff-g*}` ones?
```

Comment:
This seems a little bit concerning. Something worth asking the authors directly?

Reply:
I did, about a week ago (along with sending them the list of all the other new typos). No response yet.

I guess it's not concerning for them since it's an extra check, not a missing check, so it doesn't compromise the security. But it is concerning for me since I can't test it :)

Reply:
Also, it sort of hinges on whether we're interested in provable aborts. The authors always talk about identifiable aborts only, which don't need echo-broadcasting of those values, so these checks may be needed to make the `П^{aff-g*}` aborts identifiable. But my understanding is that we do want provable aborts (which I questioned in https://github.com/entropyxyz/manul/issues/39), so I'm operating based on that.